__tests__/app.test.js

```diff
@@ -169,6 +170,39 @@ describe("GET /api/articles?sort_by=votes / article_id/created_at/comment_count"
     })
 })
 
+describe("GET /api/articles?topic=mitch / other topics are available",()=>{
+    test("200: /api/articles?topic=mitch should return an array with all articles about the topic mitch",()=>{
+        return request(app).get("/api/articles?topic=mitch").expect(200)
+        .then(({body})=>{
+            expect(body.articles.length).toBe(12)
+            body.articles.forEach((article)=>{
+                expect(article.topic).toBe("mitch")
+            })
+        })
+    })
+    test("404: /api/articles?topic=banana should return a not found if topic does not exist in list of topics",()=>{
+        return request(app).get("/api/articles?topic=banana").expect(404)
+        .then(({body})=>{
+            expect(body.msg).toBe("No articles found. Please try a different search")
+        })
+    })
+})
+
+describe("GET /api/articles?order=asc / desc",()=>{
+    test("200: /api/articles?order=asc should return all articles in ascending order", ()=>{
+        return request(app).get("/api/articles?order=asc").expect(200)
+        .then(({body})=>{
+            expect(body.articles).toBeSortedBy("created_at",{ascending:true})
+        })
+    })
+    test("400: /api/articles?order=banana should return a 400 bad request as it is not a valid order",()=>{
+        return request(app).get("/api/articles?order=banana").expect(400)
+        .then(({body})=>{
+            expect(body.msg).toBe("Bad Request invalid order")
+        })
+    })
+})
```

Comment:
Three more scenarios to envision:
-What if I sort_by a valid column: ?sort_by=votes
-What if I sort_by an invalid column: ?sort_by=bananas
-What if I query a topic that exists but has not articles: ?topic=paper

Reply:
the first two are already tested for in lines 158 - line 170

Reply:
Ah! For some reason they're not actually a part of this PR - so they were in hidden view! Strange 👻 It's all good here anyway!